iotdb-core/datanode/src/main/java/org/apache/iotdb/db/utils/DateTimeUtils.java

```diff
@@ -733,6 +741,7 @@ public static TimeUnit toTimeUnit(String t) {
    * @return nextStartTime
    */
   public static long calcIntervalByMonth(long startTime, long numMonths) {
+    numMonths = numMonths / MS_TO_MONTH;
```

Comment:
why we need this line? numMonths shouldn't divide `MS_TO_MONTH` again.

Reply:
Because I changed function `org.apache.iotdb.db.queryengine.execution.aggregation.timerangeiterator.TimeRangeIteratorFactory#getTimeRangeIterator` , previous  code is  `interval = isIntervalByMonth ? interval / MS_TO_MONTH : interval;` while the interval is '1mo1d' to long now  ,then the input **numMonths** in **calcIntervalByMonth** is the long value **fixedSlidingStepInMonth** in new logic